### PR TITLE
refactor(demo): Repaired broken layers, added typing, removed unused imports/variables

### PR DESCRIPTION
**Pull request by LAMM26**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines:https://github.com/infra-geo-ouverte/igo2/blob/master/.github/CONTRIBUTING.md#git-commit-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Some sections of the "geo" demo had broken tools or layers.

**What is the new behavior?**
Layers have been repaired, typing has been added for claiity, tools (ie. print) have been repaired, imports/varaibles have been cleaned.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**

**Other information**:

### CODE REVIEW

File: projects/demo/src/app/geo/catalog/catalog.component.ts [hunk 1]
```diff
@@ -47,15 +48,17 @@ export class AppCatalogComponent implements OnInit {
...
       .createAsyncLayer({
         title: 'OSM',
+        baseLayer: true,
+        visible: true,
         sourceOptions: {
           type: 'osm'
         }
-      })
-      .subscribe((layer) => this.map.addLayer(layer));
+      } as LayerOptions)
```
Comment (alecarn) [inline projects/demo/src/app/geo/catalog/catalog.component.ts:60]: On essaie autant que possible d'utiliser l'opérateur [`satisfies`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html) au lieu du `as`. Parce que ce dernier force le type et empêche Typescript de faire sa validation du type.

**Review by alecarn — CHANGES_REQUESTED** [review]

**Review by alecarn — APPROVED** [review]
Look good to me, good job

**Review by aziz-access — APPROVED** [review]

File: projects/demo/src/app/geo/layer/layer.component.ts [hunk 2]
```diff
@@ -153,69 +125,42 @@ export class AppLayerComponent {
...
-    this.layerService
-      .createAsyncLayer({
-        title: 'nurc:Arc_Sample_Parent',
-        visible: false,
-        legendOptions: {
-          // collapsed: false,
-          display: true,
-          // url: 'https://v.seloger.com/s/width/1144/visuels/0/m/l/4/0ml42xbt1n3itaboek3qec5dtskdgw6nlscu7j69k.jpg',
-          stylesAvailable: [
-            { name: 'rain', title: 'Pluie' },
```
Comment (pelord) [inline projects/demo/src/app/geo/layer/layer.component.ts:200]: Cette démo était intéressante pour la définition des styles disponible pour les wms. Une alternative?

File: projects/demo/src/app/geo/search/search.component.ts [hunk 3]
```diff
@@ -89,50 +88,49 @@ export class AppSearchComponent implements OnInit, OnDestroy {
...
         sourceOptions: {
           type: 'osm'
         }
-      })
-      .subscribe((layer) => {
-        this.osmLayer = layer;
-        this.map.addLayer(layer);
-      });
+      } satisfies LayerOptions)
+      .subscribe((layer: ImageLayer) => this.map.addLayer(layer));
```
Comment (pelord) [inline projects/demo/src/app/geo/search/search.component.ts:97]: Sould be TileLayer

**Review by pelord — CHANGES_REQUESTED** [review]
Voir les commentaires.

File: projects/demo/src/app/geo/layer/layer.component.ts (hunk 2 above)
Comment (pelord) [reply projects/demo/src/app/geo/layer/layer.component.ts:200]: @LAMM26 p-etre refaire un pr pour cette demo?

**Review by pelord — APPROVED** [review]
faire un pr pour la demo de stylesavailable svp.

File: projects/demo/src/app/geo/layer/layer.component.ts (hunk 2 above)
Comment (LAMM26) [reply projects/demo/src/app/geo/layer/layer.component.ts:200]: See https://github.com/infra-geo-ouverte/igo2-lib/pull/1581

**Pull request merged**